### PR TITLE
Porting field_report to drupal console field:info

### PR DESCRIPTION
**Pull request by stevepurkiss**
Here's my first attempt, doesn't look great yet, but does work...
I felt 'field:info' is a better name, then could extend with argument for content type, e.g. case_study (a demo content type I set up)

### CODE REVIEW

**Comment by stevepurkiss** [comment]
field-info-output

**Comment by nalipaz** [comment]
I haven't tested this code yet, but I thought I would add a note here that there are a number of places in where Drupal's suggested coding standards for whitespace and some other things are not followed. Might be good to do a quick cleanup of this at some point.

**Comment by stevepurkiss** [comment]
Thanks, as mentioned this is a first draft which took me 1 day and is taken directly from http://dgo.to/field_report module. The idea was to get something working over the weekend at the contribution sprint.

There are a number of improvements that can be made, splitting up into different functions etc., and of course adhering to the coding standards.

I just thought I'd post a draft in case anyone wanted to use the functionality of the command in a purely playful way until the code is ready to be shipped.

**Comment by stevepurkiss** [comment]
@nalipaz seen as you brought up the topic of coding standards I spent ages looking around the language files trying to see if there were any patterns for the output but it seems 99% of language has been classified under 'messages'. Do you know if there's any particular reason for this? I couldn't find any "drupal console standards" in the doc, so as you'll see from my language file I split them up into 'entity-title' and 'field-header' so anyone looking at the code would have a little more info as to what the context of the string is, but do realise this now makes it different than other language files.

Let me know if there's any direction on this and I'll move the code forward appropriately...

**Comment by nalipaz** [comment]
I think this is a question for someone like @jmolivas, I wouldn't actually know. I would and do assume it would just follow standard Drupal coding standard guidelines, but perhaps not?

**Comment by stevepurkiss** [comment]
I've just had a quick look around & can't find anything specific - there's [Interface text](https://www.drupal.org/node/604342) and [Configuration schema/metadata](https://www.drupal.org/node/1905070). It's probably more of an area for the language/translation side of things as presumably providing greater context than just 'messages' would be of help when someone is making a translation of the text.

I'll keep looking, let me know if you find anything too, otherwise it might be a good idea to set some - DrupalConsole is a slightly different use case than the web UI & would be good to have something set up - there's a limited amount of functions in the input/output DrupalStyle thing so hopefully shouldn't be too much of a task to set some up, just seems a bit of everything-in-one-bucket at the moment.

**Comment by nalipaz** [comment]
You mean like this? https://www.drupal.org/coding-standards

**Comment by jmolivas** [comment]
DrupalConsole is a Symfony Application we use PRS-2 code style for this project

You can probably use this http://cs.sensiolabs.org/ and execute 
```
php-cs-fixer fix [PATH_TO_FILE]
```

Ot this https://github.com/jmolivas/phpqa and execute on your working branch 
```
phpqa analyze --git
```

**Comment by nalipaz** [comment]
@jmolivas++

**Comment by stevepurkiss** [comment]
Wow, I've never seen that before *thud*

No, I mean distinguishing between types of messages.

I know what coding standards are, I wrote my first line of code 34 years ago so have come across them a few times, I just haven't bothered thinking about them for this yet as the code isn't anywhere near releasable, it was a proof of concept & I was planning on spending some time today improving it instead of replying to pedantic responses to my contributions.

**Comment by nalipaz** [comment]
Sorry if you found my reply pedantic @stevepurkiss. I only meant to suggest that it be noted for the future as I wrote in my first comment. You pinged me in gitter so I gave what input I could with little time I have to test this for you at the moment. BTW, I am not a maintainer on this project, only a user and plan to try to start contributing how I can. Best of luck to you.

**Comment by stevepurkiss** [comment]
@jmolivas thanks for the link (my reference above was to the drupal coding standards) - I'll be sure to put the code through checks at some point, I thought my phpStorm was set up for that but obviously not!

My question was more of the text standards, table headers etc. didn't really seem logical to classify as 'messages'. That's something no automatic code checker will check but is important to know some context - I see there's 'questions' etc. but nothing much for tables that I could find.

**Comment by stevepurkiss** [comment]
@nalipaz I'm fully aware I totally over-reacted to your valid point, I didn't realise I'd pinged you on gitter, I just think we need to be aware of how we treat those who like yourself give up time to work on this stuff. Generally it's a good idea to lead in with a little praise ("oh this is cool/thanks for this/etc.") then point out your issue and provide answers and/or patches. 

To me, your comment just sounded a little brash after I'd spent my own time and money getting to a sprint so I could find out how I could help out with DrupalConsole as I'd been wanting to for a while but is never easy online and much quicker face-to-face, plus I'd stayed up till 1am getting my other laptop set up so I could get a good screenshot of the output.

Anyways, I'll get on with the task at hand - personally I'm more interested in splitting up the one big function & understanding a little more of the code I copied from the field_report module, specifically why this is done:

`$entityLabel = $entityValue->get('label');
 $labelArray = (array)$labelTest;
 $bundleParent = reset($labelArray);`

**Comment by jmolivas** [comment]
@stevepurkiss we extended the `SymfonyStyle` with the `DrupalStyle` we can keep adding m,ore features is needed.

Related to the YAML file structure. I saw your commit contains 
```
description: 'View information about fields.'
help: 'View information about fields.'
entity-title:
  block_content: 'Blocks'
  comment: 'Comments'
  contact_message: 'Contact Forms'
  node: 'Content Types'
  shortcut: 'Shortcut Menus'
  taxonomy_term: 'Taxonomy Terms'
field-header:
  name: 'Field label'
  type: 'Field type'
  desc: 'Field description'
  usage: 'Used in'
messages:
  fields-none: 'No fields are available'
```

For commands containing questions we used to do something like grouping into `options`, `arguments`, `questions` and the rest under `messages` like this example:
```
options:
    file: 'User defined file containing commands to get executed.'
questions:
    chain-file: 'Select chain file to execute'
messages
    rest_of_the_keys:
```

Feel free to do any suggestions on the YAML definitions.

**Comment by stevepurkiss** [comment]
@jmolivas thanks, I saw the options/questions/messages, didn't find anything for tables (i.e. headers).

I guess I'll have a better idea the more I work with DrupalConsole, I'll check over with what the language/translation side of things are doing as well.

**Comment by nalipaz** [comment]
@stevepurkiss sorry I came off as brash, unfortunately I know I do that IRL too. But it wasn't meant that way, and it is something I continually work on, so sorry.

**Comment by stevepurkiss** [comment]
@nalipaz lol I have the same issue with taking things personally when I shouldn't ;) Although I did get called a 'super-troll' by someone so guess I give as good as I get ;)

Let's have a Drupal hug ;)  Sorry for snapping back! I just want to code, I'll let the machines do what they do best & focus on the stuff they haven't figured out yet.

**Comment by nalipaz** [comment]
LGTM

**Comment by stevepurkiss** [comment]
@jmolivas FYI just composer installed cs.sensiolabs.org and the last line says:

`Package fabpot/php-cs-fixer is abandoned, you should avoid using it. Use friendsofphp/php-cs-fixer instead.`

**Comment by jmolivas** [comment]
@stevepurkiss thanks for letting me know, I was not aware of this.

**Comment by stevepurkiss** [comment]
Just a quick update - been going through the issues on the field_report module & changed the "Used In" to "Also Used In" so it doesn't display the current bundle & have [pushed the code back to the field_report module](https://www.drupal.org/node/2734469).

I've also strtoupper()'d the entity type and am looking as to how to change the display of the rest of it to make it look better, it's all a bit green & white at the moment but there seems to be limited amount of things I can do with DrupalStyle, I'll take a look at SymfonyStyle. Perhaps removing the description of the bundle will help clean things up, I guess it could always be an option if people need it.

Oh, and I'm gonna set up my phpStorm properly so it doesn't insert lots of whitespace & get the spacing wrong... ;)

File: src/Command/Field/InfoCommand.php [hunk 1]
```diff
@@ -1,10 +1,5 @@
 <?php
 
-/**
```
Comment (jmolivas) [inline src/Command/Field/InfoCommand.php]: No longer requiring `@file` is based on DrupalCoding standards this app must be PSR2 compatible

Comment (stevepurkiss) [inline src/Command/Field/InfoCommand.php]: OK cool, will get the hang of all this soon ;)

**Comment by stevepurkiss** [comment]
I've cleaned up the code a bit and changed the format - currently it's not displaying the 'full' information, i.e. the bundle and field descriptions. It's readable, so I think the full descriptions should be an option, I'll figure out how to do that.

I put the no fields in a commentBlock as this seemed the only current way to make it stand out but seems a little too much now so not sure, hence posting screenshot.

I still want to split the one big function up, especially now I've gone back to 4 spaces for indentation. I guess I find it hard to get out of my procedural programming brain ;)

Also I created (using DrupalConsole of course!) a custom entity type and a field plugin but it's not displaying the fields for that, I don't think the original module ever did that so will figure that out and push back to the field_report module.

Will continue...

field-info-output-comment-block

**Comment by stevepurkiss** [comment]
Ahah, I figured out how to run psr2 on phpcs and squashed in the latest code. 

I'll run the codesniffer *before* I commit next time, soz ;)

**Comment by jmolivas** [comment]
@stevepurkiss: This looks great. Less info but way more easy to read.

> so I think the full descriptions should be an option

We can probably add an option `--detailed` to show the full report including more columns

**Comment by stevepurkiss** [comment]
Thanks, yes indeed, an option later, depends on when you want/need it to go upstream, I'll be on the case again tomorrow, getting my coding chops back in action!

It's still not working for custom entities yet AFAIK, I'm not entirely convinced my custom entity is created correctly, not had chance to delve into that yet, but will do before the --detailed option as presume some users will have custom entities. There's probably a module out there with custom entities I can use to check - off top of my head something like profile or whatever if that's around for D8.

Meanwhile, it's now cidertime @purkisshq ;)

**Comment by stevepurkiss** [comment]
Yay, it *does* work for custom entities, I just hadn't set mine up right ;) 

I added two bundles to my Purki Entity and a boolean field on the first, no fields on the second and it picked them up OK:

**Comment by stevepurkiss** [comment]
Oops, bit too trigger-happy there, here you go:

field-info-output-custom-entities

**Comment by stevepurkiss** [comment]
Squashing commits as improvements to be made but code currently in working state.

**Comment by stevepurkiss** [comment]
Just adding a note to say thanks to @jmolivas the help text is now working yay!

Am working on arguments now so can restrict to specific entity, and possibly bundle.

**Comment by stevepurkiss** [comment]
OIC, options, not arguments. Will understand more soon :D

**Comment by jmolivas** [comment]
@stevepurkiss let me know if you need any help pinging me he here or at gitter channel.

**Comment by stevepurkiss** [comment]
@jmolivas thanks - I haven't fully figured out how it'll work yet but thought restricting by entity type(s) and/or bundle would be useful, e.g. show me all the fields for taxonomy term entities; show me all the fields for article content type/bundle. 

The issue is you could say show me all the fields for X entity AND Y bundle where they don't match but I guess it could just say that's not a valid input.

I'm still pondering best approach, mostly been RTFM on Symfony Console today - and cooking a roast dinner round my mum's ;)

**Comment by stevepurkiss** [comment]
Ooops, something tells me I didn't do that right, was trying to update my local branch after fetching upstream then merging upstream/master then rebasing onto my local branch. Ignore pls whilst I re-remember rebasing etc. ;)

**Comment by stevepurkiss** [comment]
Continuing over at PR #2572

**Pull request closed without merging**